src/nginx_config.py

```diff
@@ -350,6 +351,9 @@ def _server(
                     {"directive": "ssl_certificate_key", "args": [KEY_PATH]},
                     {"directive": "ssl_protocols", "args": ["TLSv1", "TLSv1.1", "TLSv1.2"]},
                     {"directive": "ssl_ciphers", "args": ["HIGH:!aNULL:!MD5"]},  # pyright: ignore
+                    # specify resolver to ensure that if a unit IP changes,
+                    # we reroute to the new one
+                    *self._resolver(custom_resolver=_get_dns_ip_address()),
```

Comment:
Here it's only added when TLS is on. We probably need it also when TLS is not there.